packages/cache-manager/src/index.ts

```diff
@@ -122,11 +124,28 @@ export const createCache = (options?: CreateCacheOptions): Cache => {
 	};
 
 	const mget = async <T>(keys: string[]): Promise<Array<T | undefined>> => {
-		const result: Array<T | undefined> = [];
+		let result: Array<T | undefined> = keys.map(() => undefined);
 
-		for (const key of keys) {
-			const data = await get<T>(key);
-			result.push(data);
+		if (nonBlocking) {
+			try {
+				result = await Promise.race(stores.map(async store => store.getMany(keys)));
+				eventEmitter.emit('mget', {keys, values: result});
+			} catch (error) {
+				eventEmitter.emit('mget', {keys, error});
+			}
+		} else {
+			for (const store of stores) {
+				try {
+					const cacheValue = await store.getMany<T>(keys);
```

Comment:
I wasn't sure if it should take the first successful response entirely or if it should go to each store and merge the results if an earlier store didn't return something

Reply:
for nonBlocking you should take the first store response. I believe the pattern that we have been going with for normal mode is we loop through all of them and merge results. Correct?

Reply:
Pushed that change and fixed the lint error